x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_error_log.tsx

```diff
@@ -0,0 +1,275 @@
+        ?.map(({ from, to, display }: { from: string; to: string; display: string }) => ({
+          start: from,
+          end: to,
+          label: display,
+        })) || []
+    );
+  });
+
+  const isInitialized = useRef(false);
+
+  const loadEventLogs = async () => {
+    setIsLoading(true);
+    try {
+      const result = await loadExecutionLogAggregations({
+        id: rule.id,
+        sort: {
+          [sort?.field || 'timestamp']: { order: sort?.direction || 'desc' },
+        } as unknown as LoadExecutionLogAggregationsProps['sort'],
+        dateStart: getParsedDate(dateStart),
+        dateEnd: getParsedDate(dateEnd),
+        page: 0,
+        perPage: 1,
+      });
+      setLogs(result.errors);
+      const start = pagination.pageIndex * pagination.pageSize;
+      const logsSortDesc = result.errors.sort((a, b) => sortErrorLog(a, b, sort?.direction));
+      setLogList(logsSortDesc.slice(start, start + pagination.pageSize));
+      setPagination({
+        ...pagination,
+        totalItemCount: result.totalErrors,
+      });
+    } catch (e) {
+      notifications.toasts.addDanger({
+        title: API_FAILED_MESSAGE,
+        text: e.body.message,
+      });
+    }
+    setIsLoading(false);
+  };
+
+  const onTimeChange = useCallback(
+    ({ start, end, isInvalid }: OnTimeChangeProps) => {
+      if (isInvalid) {
+        return;
+      }
+      setDateStart(start);
+      setDateEnd(end);
+    },
+    [setDateStart, setDateEnd]
+  );
+
+  const onRefresh = () => {
+    loadEventLogs();
+  };
+
+  const columns: Array<EuiBasicTableColumn<IErrorLog>> = useMemo(
+    () => [
+      {
+        field: 'timestamp',
+        name: i18n.translate(
+          'xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.timestamp',
+          {
+            defaultMessage: 'Timestamp',
+          }
+        ),
+        render: (date: string) => (
+          <RuleEventLogListCellRenderer columnId="timestamp" value={date} dateFormat={dateFormat} />
+        ),
+        sortable: true,
+        width: '250px',
+      },
+      {
+        field: 'type',
+        name: i18n.translate('xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.type', {
+          defaultMessage: 'Type',
+        }),
+        sortable: false,
+        width: '100px',
+      },
+      {
+        field: 'message',
+        name: i18n.translate(
+          'xpack.triggersActionsUI.sections.ruleDetails.errorLogColumn.message',
+          {
+            defaultMessage: 'Message',
+          }
+        ),
+        sortable: false,
+      },
+    ],
+    [dateFormat]
+  );
+
+  useEffect(() => {
+    loadEventLogs();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dateStart, dateEnd]);
+
+  useEffect(() => {
+    const start = pagination.pageIndex * pagination.pageSize;
+    setLogList(logs.slice(start, start + pagination.pageSize));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pagination]);
```

Comment:
what do you think about making this a memoized array so we don't have to duplicate this and in the fetch handler

Reply:
good idea!!!